tests/foreman/api/test_classparameters.py

```diff
@@ -47,7 +47,7 @@ def invalid_sc_parameters_data():
         {'sc_type': 'real', 'value': gen_string('alpha')},
         {'sc_type': 'array', 'value': '0'},
         {'sc_type': 'hash', 'value': 'a:test'},
-        {'sc_type': 'yaml', 'value': '{a:test}'},
+        {'sc_type': 'yaml', 'value': "@see".strip('"')},
```

Comment:
```suggestion
        {'sc_type': 'yaml', 'value': '@see'},
```
The `strip` doesn't make sense to me, unless I'm missing something?
The double quotes defines where the string begins and ends and there is no double quote in the string itself.
Wouldn't this be the same?
```
In [1]: {'sc_type': 'yaml', 'value': "@see".strip('"')}
Out[1]: {'sc_type': 'yaml', 'value': '@see'}

In [2]: {'sc_type': 'yaml', 'value': "@see"}
Out[2]: {'sc_type': 'yaml', 'value': '@see'}

In [3]: {'sc_type': 'yaml', 'value': '@see'}
Out[3]: {'sc_type': 'yaml', 'value': '@see'}
```

Reply:
@vsedmik I added `"@see".strip('"')` to prevent any typos in the test. When the value is added to the YAML parameter, this removes the quotes, converting the value from a string to invalid value. For the negative test, I need to provide an invalid value. If you pass anything as a string for the YAML parameter, it will save the value, which is not the desired behavior for a negative test.

Reply:
@amolpati30  I'm sorry but I'm afraid I still don't get the point. Let me explain: The `invalid_sc_parameters_data` function returns list of dictionaries - one item for each parametrized run. Inside the test you get just one dictionary like `{'sc_type': 'yaml', 'value': '@see'}` through the `test_data` parameter and you access its values [here](https://github.com/vsedmik/robottelo/blob/3f11c65dfd5ddb880c526e04a97f1430f17ce2d0/tests/foreman/api/test_classparameters.py#L137-L138).

The values of the `sc_type` and `value` keys of the passed dictionary are of the `string` type. The value of the `value` key is just `@see`, which is really desired _invalid YAML_ syntax and the `sc_param.update` should raise the HTTPerror. My point is that it does not matter if you define string as `'@see'` or `"@see"` or `'''@see'''` or `"""@see"""` - the value never contains the `"` inside, so the strip is useless.

Reply:
@vsedmik 
If we pass a string-type value like `{'sc_type': 'yaml', 'value': '@see'}`, it will not raise pytest.raises(HTTPError) because YAML accepts the string value, which is not desirable for the test. However, we cannot pass the value without enclosing it in a string (key-value pair) as it would result in a syntax error.

To handle this, the value for the key can be modified as follows:
`{'sc_type': 'yaml', 'value': "@see".strip('"')}`
When this value is passed to `sc_param.update()` for YAML, it will raise an error, making the invalid test work as expected.

Regarding your second point, as you mentioned, if we pass the value as '@see', the value never contain (") inside. Therefore, passing the value '@see' would still be interpreted as a string in Satellite and would be accepted.

To address this, I removed the quotes around the value and tested it, which resolved the issue. If you're still unsure, please try the test with '@see' values for the YAML parameter to identify one that triggers the desired error. If you are not agree with this so can you please provide any invalid value for yaml param.

Reply:
@amolpati30 I'm afraid I have to disagree. The value you are passing inside the test is _always_ `{'sc_type': 'yaml', 'value': '@see'}`, no matter if you do the `strip` or not, because the `strip` actually does nothing in this case. The value of `value` is always type of string, but the key is _what_ is inside the string.

The reason why the test passes now is because `@see` is truly invalid YAML format, while the original `{a:test}` _is valid_ YAML format. You can try https://yamlchecker.com/ or other format checker if you want.

I took the effort, set up the `session_puppet_enabled_sat` and tested locally. In both cases (strip and no strip) the test passed the same since the strip is redundant. We can do a quick debug session if you want, I still have the setup ready.

Reply:
@vsedmik Thanks! It is working now.